pyfr/solvers/euler/kernels/flux.mako

```diff
@@ -24,3 +24,40 @@
     f[${i}][${j + 1}] = rhov[${i}]*v[${j}]${' + p' if i == j else ''};
 % endfor
 </%pyfr:macro>
+
+<%pyfr:macro name='inviscid_flux_1d' params='s, f, p, v'>
+    fpdtype_t invrho = 1.0/s[0], E = s[${nvars - 1}];
+
+    // Compute the velocities
+% for i in range(ndims):
+    v[${i}] = invrho*s[${i + 1}];
+% endfor
+
+    // Compute the pressure
+    p = ${c['gamma'] - 1}*(E - 0.5*invrho*${pyfr.dot('s[{i} + 1]', i=ndims)});
+
+    // Density and energy fluxes
+    f[0] = s[1];
+    f[${nvars - 1}] = (E + p)*v[0];
+
+    // Momentum fluxes
+    f[1] = s[1]*v[0] + p;
+% for j in range(1, ndims):
+    f[${j + 1}] = s[1]*v[${j}];
+% endfor
+</%pyfr:macro>
+
+<%pyfr:macro name='inviscid_pri' params='s, p, v, c'>
+    fpdtype_t invrho = 1/s[0], E = s[${nvars - 1}];
+
+    // Compute the velocities
+% for i in range(ndims):
+    v[${i}] = invrho*s[${i + 1}];
+% endfor
+
+    // Compute the pressure
+    p = ${c['gamma'] - 1}*(E - 0.5*invrho*${pyfr.dot('s[{i} + 1]', i=ndims)});
```

Comment:
{i + 1}

Reply:
I don't think you can do that here. At least it gives an error. This is also the syntax used here: https://github.com/PyFR/PyFR/blob/78884175acf239b588e7b812634ef8bb1adb8094/pyfr/solvers/euler/kernels/entropy.mako#L12

Reply:
I think you can use the syntax `p = ${c['gamma'] - 1}*(E - 0.5*invrho*${pyfr.dot('s[{i} + 1]', i=(1, ndims+1))});` but I'm not 100% sure